packages/block/src/header.ts

```diff
@@ -132,6 +135,20 @@ export class BlockHeader {
       },
     ]
     defineProperties(this, fields, data)
+
+    if (this._common.hardforkIsActiveOnChain('dao')) {
+      // verify the extraData field.
+      const blockNumber = new BN(this.number)
+      const DAOActivationBlock = new BN(this._common.hardforkBlock('dao'))
+      if (blockNumber.gte(new BN(DAOActivationBlock))) {
```

Comment:
Regarding why we are using BNs in block numbers: I think there might be some ethereum/tests case where it could use a big number? The test cases test some pretty extreme things (which is OK of course if they are valid within the protocol).

Reply:
I will keep it a BN; this allows for flexibility in case someone wants to ever use a gigantic block number for the DAO transition.